lib/engine/part/path.rb

```diff
@@ -5,16 +5,18 @@
 module Engine
   module Part
     class Path < Base
-      attr_reader :a, :b, :branches, :city, :edges, :junction, :nodes,
-                  :offboard, :stops, :terminal, :town
-      def initialize(a, b, terminal: nil)
+      attr_reader :a, :b, :branches, :city, :edges, :junction, :lanes, :lane_index,
+                  :nodes, :offboard, :stops, :terminal, :town
+      def initialize(a, b, terminal: nil, lanes: nil, lane_index: 0)
         @a = a
         @b = b
         @edges = []
         @branches = []
         @stops = []
         @nodes = []
         @terminal = !!terminal
+        @lanes = lanes || 1
+        @lane_index = lane_index
```

Comment:
i wonder if we can make lane_index inferred, it sucks that you have to pass @lanes and @lane_index

perhaps if parallel? you can just return the index of the particular path as a helper method

Reply:
I don't understand your solution. We need two attributes for a path: how many lanes in parallel and what is the position within those lanes, so both @lanes and @lane_index need to be stored with a path.

Reply:
can't you infer the lane_index based on the position in the array?

Reply:
Which array? There are potentially many paths on a tile, not all of them part of parallel paths.